test/test_matmul_cores/test_trig.cc

```diff
@@ -0,0 +1,645 @@
+                    blas::gemm(
+                        blas::Layout::RowMajor,
+                        blas::Op::Trans,
+                        blas::Op::NoTrans,
+                        m,
+                        n,
+                        m,
+                        1.0,
+                        perm_matrix,
+                        m,
+                        A_vec.data(),
+                        n,
+                        0.0,
+                        true_perm_matrix,
+                        n
+                    );
+                    RandBLAS::permute_rows_to_top(layout, m, n, v, 1, A_vec.data());
+                }
+                else {
+                    blas::gemm(
+                        blas::Layout::ColMajor,
+                        blas::Op::Trans,
+                        blas::Op::NoTrans,
+                        m,
+                        n,
+                        m,
+                        1.0,
+                        perm_matrix,
+                        m,
+                        A_vec.data(),
+                        m,
+                        0.0,
+                        true_perm_matrix,
+                        m
+                    );
+                    RandBLAS::permute_rows_to_top(layout, m, n, v, 1, A_vec.data());
+                }
+            }
+            else {
+                if(layout == blas::Layout::RowMajor) {
+                    blas::gemm(
+                        blas::Layout::RowMajor,
+                        blas::Op::NoTrans,
+                        blas::Op::NoTrans,
+                        m,
+                        n,
+                        n,
+                        1.0,
+                        A_vec.data(),
+                        n,
+                        perm_matrix,
+                        n,
+                        0.0,
+                        true_perm_matrix,
+                        n
+                    );
+                    RandBLAS::permute_cols_to_left(layout, m, n, v, 1, A_vec.data());
+                }
+                else {
+                    blas::gemm(
+                        blas::Layout::ColMajor,
+                        blas::Op::NoTrans,
+                        blas::Op::NoTrans,
+                        m,
+                        n,
+                        n,
+                        1.0,
+                        A_vec.data(),
+                        m,
+                        perm_matrix,
+                        n,
+                        0.0,
+                        true_perm_matrix,
+                        m
+                    );
+                    RandBLAS::permute_cols_to_left(layout, m, n, v, 1, A_vec.data());
+                }
+            }
+
+            T norm_permute = 0.0;
+            blas::axpy(m * n, -1.0, A_vec.data(), 1, true_perm_matrix, 1);
+            norm_permute = blas::nrm2(m * n, true_perm_matrix, 1);
+
+            randblas_require(norm_permute < tol);
+
```

Comment:
Don't check for array equality by checking if norm is (sufficiently) small. Check componentwise so we get more information if/when a test fails. Use
https://github.com/BallisticLA/RandBLAS/blob/fd180c5743757936fdfa5be8b0fecc62d66b55c4/test/comparison.hh#L181-L229